lab_5_scrapper/scrapper.py

```diff
@@ -1,82 +1,205 @@
 """
 Crawler implementation
 """
-from typing import Pattern, Union
+import datetime
+import json
+import locale
+import random
+import re
+import shutil
+import time
+from pathlib import Path
+from typing import List, Dict, Pattern, Union
+from urllib.parse import urlparse
+
+import requests
+from bs4 import BeautifulSoup
+
+from core_utils.article.article import Article
+from core_utils.article.io import to_raw, to_meta
+from core_utils.config_dto import ConfigDTO
+from core_utils.constants import ASSETS_PATH, \
+                                                                    CRAWLER_CONFIG_PATH, \
+                                                                    NUM_ARTICLES_UPPER_LIMIT, \
+                                                                    TIMEOUT_LOWER_LIMIT, TIMEOUT_UPPER_LIMIT
+
+
+class IncorrectSeedURLError(Exception):
+    """ Inappropriate value for seed url"""
+
+
+class NumberOfArticlesOutOfRangeError(Exception):
+    """ Number of articles either to large or small"""
+
+
+class IncorrectNumberOfArticlesError(Exception):
+    """ Inappropriate value for number of articles"""
+
+
+class IncorrectHeadersError(Exception):
+    """ Inappropriate value for headers"""
+
+
+class IncorrectEncodingError(Exception):
+    """ Inappropriate value for encoding"""
+
+
+class IncorrectTimeoutError(Exception):
+    """ Inappropriate value for timeout"""
+
+
+class IncorrectVerifyError(Exception):
+    """ Inappropriate value for certificate"""
 
 
 class Config:
     """
     Unpacks and validates configurations
     """
 
+    _seed_urls: List[str]
+    _num_articles: int
+    _headers: Dict[str, str]
+    _encoding: str
+    _timeout: int
+    _headless_mode: bool
+    _should_verify_certificate: bool
+
     def __init__(self, path_to_config: Path) -> None:
         """
         Initializes an instance of the Config class
         """
-        pass
+        self.path_to_config = path_to_config
+        self._validate_config_content()
+        self._extract_config_content()
 
     def _extract_config_content(self) -> ConfigDTO:
         """
         Returns config values
         """
-        pass
+        with open(self.path_to_config, 'r', encoding='utf-8') as f:
+            config_dict = json.load(f)
+        self._seed_urls = config_dict['seed_urls']
+        self._num_articles = config_dict['total_articles_to_find_and_parse']
+        self._headers = config_dict['headers']
+        self._encoding = config_dict['encoding']
+        self._timeout = config_dict['timeout']
+        self._should_verify_certificate = config_dict['should_verify_certificate']
+        self._headless_mode = config_dict['headless_mode']
+
+        config_instance = ConfigDTO(seed_urls=self._seed_urls,
```

Comment:
first you create DTO, return it from this method.

as it is called in init, you assign self.ATTR_NAME to values taken from that DTO